### ISSUE TITLE
Support for Hazard Events [HC API]

### ISSUE THREAD

**Issue by p-a-s-c-a-l** [clarity-h2020/emikat#11]
This is related to the [production of Heat Wave Hazard Events for Europe](https://github.com/clarity-h2020/csis/issues/57).

While hazard indexes are currently stored on [METEOROID Geoserver](https://clarity.meteogrid.com/geoserver/web/) and accessed by the [Map Component](https://github.com/clarity-h2020/map-component/issues/33) for informative purposes, we need another API that is accessed by the [HC Events Table]() and the [Scenario Selection UI](https://github.com/clarity-h2020/docker-drupal/issues/58).

The API would expect a parameter '$studyArea' or, if the study is already [initialised](https://github.com/clarity-h2020/emikat/issues/6), the '$studyId' (EMIKAT internal id) and additional a parameter `$hazardEventType` (name/id from the CSIS [Hazards Taxonomy](https://csis.myclimateservice.eu/admin/structure/taxonomy/manage/hazards/overview)).

It's currently unclear, if it would make sense to show HC Events Layers in the [HC Map](https://github.com/clarity-h2020/map-component/issues/33). Therefore there is *currently* no need for an additional WMS Endpoint.

The respective [HC Data Package Resource](https://github.com/clarity-h2020/data-package/issues/8#issuecomment-506659185) has to be updated accordingly and example request / response URLs have to be [added to the Wiki](https://github.com/clarity-h2020/csis/issues/61).

**Comment by p-a-s-c-a-l** [comment]
See https://github.com/clarity-h2020/csis/issues/57#issuecomment-509101067

**Comment by bernhardsk** [comment]
The available requests are documented via Swagger:
https://service.emikat.at/EmiKatTst/swagger/index.html

Some of the relevant functionalities:
![grafik](https://user-images.githubusercontent.com/42140898/61233918-01170500-a732-11e9-881b-6bfb5fc405bd.png)

**1) For Europe wide Hazard Layers (currently: heat wave only):**
Table in AIT EMIKAT:
![grafik](https://user-images.githubusercontent.com/42140898/61233613-366f2300-a731-11e9-8e72-fc43167c9aa2.png)

Requests:
![grafik](https://user-images.githubusercontent.com/42140898/61234076-771b6c00-a732-11e9-8ecb-a89c2e2244a1.png)

JSON:
https://service.emikat.at/EmiKatTst/api/scenarios/2846/feature/tab.CLY_HAZARD_EVENTS_EUROPE.2036/table/data?rownum=1000

2846 = internal studyID (in emikat: scenarioID).
"tab.CLY_HAZARD_EVENTS_EUROPE.2036" = emikat featureName
For the first 1000 datasets. Total number of datasets in this table: 1210950

CSV: 
https://service.emikat.at/EmiKatTst/api/scenarios/2846/feature/tab.CLY_HAZARD_EVENTS_EUROPE.2036/table/csv?rownum=1000

**2) For regional Hazard Event datasets:**
Table in AIT EMIKAT:
![grafik](https://user-images.githubusercontent.com/42140898/61235005-c19de800-a734-11e9-9e23-61240e4a4e0d.png)

JSON: 
https://service.emikat.at/EmiKatTst/api/scenarios/2846/feature/tab.CLY_HAZARD_EVENTS_STUDY.2036/table/data?rownum=1000

CSV: 
https://service.emikat.at/EmiKatTst/api/scenarios/2846/feature/tab.CLY_HAZARD_EVENTS_STUDY.2036/table/csv?rownum=1000

Also available (in general): GEOJSON and GEOTIFF. ...."/table/geojson?rownum=1000" or: ....."/table/geotiff"

Please let me know, if additional info is required.

**Comment by p-a-s-c-a-l** [comment]
Thanks @bernhardsk 

Two more questions:
1) *"The API would expect a parameter `$studyArea` or `$studyId`*" 
This is currently not supported? We need the events relevant for the selected study area (bounding box). This bounding box is IMHO stored in EMIKAT when EMIKAT [is notified about a new study](https://github.com/clarity-h2020/emikat/issues/6). So an additional parameter `$studyId` (the EMIKAT internal id of the study) should be sufficient. However, the scenario id is already part of the URI (.../scenarios/**2846**/...), right? So the filtering by study area is not yet implemented?
2) The table layout will change when the [new Hazard Events](https://github.com/clarity-h2020/csis/issues/57) are available. So the new table structure would look like this, right?:

| latitude  | longitude | urban area | country | RCP      | Period            | Tx95p | Frequent(1.0) | Occasional(0.2) | Rare(0.05) |
|-----------|-----------|------------|---------|----------|-------------------|-------|---------------|-----------------|------------|
| 30.055091 | -8.120772 |            |         | baseline | 19710101-20001231 | 39C   | 36C           | 38C             | 40C        |

**Comment by p-a-s-c-a-l** [comment]
FYI: [Here](https://github.com/clarity-h2020/csis/issues/57) are the heat wave events for the baseline climate for all of Europe.

**Comment by humerh** [comment]
The call

https://service.emikat.at/EmiKatTst/api/scenarios/2846/feature/tab.CLY_HAZARD_EVENTS_STUDY.2036/table/data?rownum=1000

returns now all "relevant" events inside of study area.

The id "2846" has to be replaced by the $emikat_id .

**Comment by p-a-s-c-a-l** [comment]
ATM the API events for RCP45 only. Did you import all files from https://github.com/clarity-h2020/csis/issues/57#issuecomment-516723092 ?

**Comment by humerh** [comment]
All events are now imported.

**Comment by p-a-s-c-a-l** [comment]
No Hazard Data seems to be available for [this study](
https://csis.myclimateservice.eu/study/35/step/1454/view/table) (AgiosDimitrios): https://service.emikat.at/EmiKatTst/api/scenarios/3189/feature/tab.CLY_HAZARD_EVENTS_STUDY.2036/table/data?rownum=1000

**Comment by p-a-s-c-a-l** [comment]
Same for [this study](https://csis.myclimateservice.eu/study/33/step/1460/view/table) (ALBA_IULIA): https://service.emikat.at/EmiKatTst/api/scenarios/3209/feature/tab.CLY_HAZARD_EVENTS_STUDY.2036/table/data?rownum=1000

**Comment by DenoBeno** [comment]
maybe the input data is still not on atos server?

**Comment by p-a-s-c-a-l** [comment]
Then this would just affect [HC-LE](https://github.com/clarity-h2020/emikat/issues/12#issuecomment-534932804) and not HC.

**Comment by humerh** [comment]
There are no "LAND_USE" date (Urban atlas data) available, so Emikat can not calculate.

**Comment by humerh** [comment]
If you want a support to calculate each location in Europe we have to define for all cities also the location specific parameters.

As an example:
![image](https://user-images.githubusercontent.com/20284964/65608509-a0683c00-dfae-11e9-9dfb-6a007f17530d.png)

**Comment by p-a-s-c-a-l** [comment]
> There are no "LAND_USE" date (Urban atlas data) available, so Emikat can not calculate.

Cannot calculate what? HC or HC-LE. I thought HC (events) is independent of the HC-LE Urban Atlas input layers. So at least the EE and HC data should be available for whole Europe.

@negroscuro Can you please check if HC-LE input data is available for the following study areas: [Alba Iulia](https://csis.myclimateservice.eu/study/33/view/area) and [Athens](https://csis.myclimateservice.eu/study/35/view/area). I could select both cities from the cities taxonomy, so I assume that a complete set of HC-LE input layers is available for those? Maybe they haven't been generated by your script yet?

**Comment by humerh** [comment]
Sorry, now I understand.
Yes, I had a simlified implementation of the script for searching the "relevant events" for the study area. 
I filtered the table for  the city names, which was placticable for NAPOLI and STOCKHOLM.
But the table of Robert had not filled all cities.
So I implemented yesterday an imporved script, which looks for the  EVENTS, which are NEAREST the middlepoint of the study area, ignoring the given city names (urban area).
This is now the functionality we decided in Frankfurt.
